test/functional/shared/mechanics/mechanics_test.go

```diff
@@ -333,16 +323,16 @@ func Test_InvalidResourceIDs(t *testing.T) {
 			Code: "ResourceDeploymentFailure",
 			Details: []step.DeploymentErrorDetail{
 				{
-					Code:            "BadRequest",
-					MessageContains: "application ID \"global\" for the resource",
+					Code:            "Internal",
+					MessageContains: "'global' is not a valid resource id",
 				},
 			},
 		},
 	})
 
 	test := shared.NewRPTest(t, name, []shared.TestStep{
 		{
-			Executor: step.NewDeployErrorExecutor(template, validate, functional.GetMagpieImage()),
+			Executor: step.NewDeployErrorExecutor(template, validate, functional.GetMagpieImage(), functional.GetBicepRecipeRegistry(), functional.GetBicepRecipeVersion()),
```

Comment:
Why do we need the additional parameters?

Reply:
Since I removed the HTTPRoutes that were used as part of test, Ryan suggested I could add in the extender. These changes were needed as part of that.